### PR TITLE
Change the automation code

### PR DESCRIPTION
**Pull request by troydai**
1. Install automation code as a development package. The outcome is that the automation code can be invoked without add ./scripts folder to the PYTHONPATH.
2. Add PEP8 scripts to fix PEP8 automatically

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,7 +2,6 @@ adal==0.4.3
 applicationinsights==0.10.0
 argcomplete==1.3.0
 colorama==0.3.7
-coverage==4.2
```
Comment (tjprescott) [inline requirements.txt]: Why are we removing this dependency?

File: scripts/automation/style/pep8.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+
+def fix_pip8(directory):
```
Comment (tjprescott) [inline scripts/automation/style/pep8.py]: typo: should be `fix_pep8`.  Also, where does it "fit" (i.e. how and when will we invoke it?) Should it be added to the CI scripts?

File: requirements.txt (hunk 1 above)
Comment (troydai) [inline requirements.txt]: It is moved to automation package's setup.py.

File: scripts/automation/style/pep8.py (hunk 2 above)
Comment (troydai) [inline scripts/automation/style/pep8.py]: 1. I'll fix the typo;
2. I don't think this should be added to the CI script. From what I saw the autopep8 doesn't fix everything perfectly. It needs human intervention.
3. What I do plan to add is a PEP8 code style check run after pylint.

File: scripts/setup.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+]
+
+DEPENDENCIES = [
+    'autopep8==1.2.4',
+    'coverage==4.2',
+    'pycodestyle==2.2.0'
+]
+
+setup(
+    name='automation',
```
Comment (derekbekoe) [inline scripts/setup.py]: As discussed, please change name  as even though we won't be releasing this package, it conflicts with https://pypi.python.org/pypi/automation.

**Comment by troydai** [comment]
Looks good? I'll merge soon.

File: scripts/setup.py (hunk 3 above)
Comment (troydai) [inline scripts/setup.py]: I'll fix this right now.

**Pull request merged**